back/node_watcher/src/tasks/createReferendumStatus.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright 2018-2020 @paritytech/nomidot authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { ApiPromise } from '@polkadot/api';
+import { BlockNumber, Hash } from '@polkadot/types/interfaces';
+import { logger } from '@polkadot/util';
+
+import { prisma } from '../generated/prisma-client';
+import {
+  NomidotReferendumRawEvent,
+  NomidotReferendumStatusUpdate,
+  referendumStatus,
+  Task,
+} from './types';
+
+const l = logger('Task: Referendum Status Update');
+
+/*
+ *  ======= Table (Proposal Status Update) ======
+ */
+const createProposal: Task<NomidotReferendumStatusUpdate[]> = {
+  name: 'createProposalStatusUpdate',
+  read: async (
+    blockHash: Hash,
+    api: ApiPromise
+  ): Promise<NomidotReferendumStatusUpdate[]> => {
+    const events = await api.query.system.events.at(blockHash);
+
+    // The "Started" event is taken care of by the createReferendum
+    // task, so we need to filter it out.
+    const referendumEvents = events.filter(
+      ({ event: { method, section } }) =>
+        section === 'democracy' &&
+        Object.values(referendumStatus)
```

Comment:
`Object.values(referendumStatus)` returns `string[]`. Here, we could cast it: `Object.values(referendumStatus) as ReferendumStatus[]`

Reply:
If I remember correctly, such a casting wasn't accepted, because the types "string" and `ReferendumStatus` have nothing in common according to TS.

I ended up using type of key of, which is not super readable at first, but I think it's nice.